File: Python/bien_doi_ve_1.py
while 1:
    n = int(input())
    if n==0: break
    if n==1: print("1")
    else:
        dem = 1
        while n!=1:
            if n%2==0: n/=2
            else: n = n*3 + 1
            dem+=1
        print(dem)

File: C++/biendoive1.cpp
#include <bits/stdc++.h>
#define ll long long
#define pb push_back
#define maxn 1000005
using namespace std;

int main(){
//By Kenzy 	v Don't copy plzzz !! Tks !!
	int t;
	cin>>t;
	while(t--){
		int n;
		cin>>n;
		int dem=0;
		while(n>0){
			if(n%3==0){
				while(n%3==0){
					n/=3;
					dem++;
				}
			}
			else if(n%2==0){
				while(n%2==0){
					n/=2;
					dem++;
				}
			}
			else{
				n--;
				dem++;
			}
			if(n==1){
				break;
			}
		}
		cout<<dem<<endl;
	}
return 0;
}

